mayavi/core/scene.py

```diff
@@ -198,6 +200,11 @@ def tno_drop_object(self, node, dropped_object):
     ######################################################################
     # Non-public interface
     ######################################################################
+    def _closed_fired(self):
+        self.scene = None
+        self.menu_helper.object = None
+        self.parent.closed = True
```

Comment:
Basically, the `SceneModel` and other subclasses have a "closing" event that is automatically hooked up to close references in `engine.new_scene`.